Documentation/Books/AQL/Graphs/KShortestPaths.md

```diff
@@ -0,0 +1,174 @@
+k Shortest Paths in AQL
+====================
+
+General query idea
+------------------
+
+This type of query is supposed to find the first *k* paths in order of length
+(or weight) between two given documents, *startVertex* and *targetVertex* in
+your graph.
+
+Every such path will be returned as a JSON object with three components:
+
+1. an array of the vertices on the path,
+2. an array of the edges on the path, and
+3. the weight of the path.
+
+If no *weightAttribute* is given, the weight of the path is just its length.
+
+### Example execution
+
+Let's take a look at a simple example to explain how it works.
+This is the graph that we are going to find some shortest path on:
+
+![train_map](train_map.png)
+
+Let us assume 
+
+1. We start at the vertex **Aberdeen**.
+2. We finish with the vertex **London**.
+
+We expect to see the vertices **Aberdeen**, **Leuchars**, **Edinburgh** and **York**
+and **London** on *the* shortest path, in this order. An alternative path, which is slightly longer
+goes from **Aberdeen**, **Leuchars**, **Edinburgh**, **York**, **Carlisle**, **Birmingham**, **London**
+
+Taking into account edge weights, the second path deviates, because the route
+**Aberdeen**, **Leuchars**, **Edinburgh**, **Glasgow**, **Carlisle**,
+**Birmingham**, **London** is quicker.
+
+Syntax
+------
+
+The syntax for k Shortest Paths queries is similar to the one for [Shortest Path](ShortestPath.md).
+You have two options: either use a named graph, or a set of edge
+collections.
+
+It is highly recommended that you use a **LIMIT** statement,
+as k Shortest Paths is an expensive operation, and on large graphs can lead to a large
+number of paths being returned, or an expensive (but unsuccessful) search for more shortest paths
+might be performed.
+
+### Working with named graphs
+
+```
+FOR path 
+  IN OUTBOUND|INBOUND|ANY K_SHORTEST_PATHS
+  startVertex TO targetVertex
+  GRAPH graphName
+  [OPTIONS options]
+```
+
+- `FOR`: the variable **path** which contains one path as an object containing 
+   vertices, edges, and the weight of the path.
+- `IN` `OUTBOUND|INBOUND|ANY`: defines in which direction edges are followed
+  (outgoing, incoming, or both)
+- **startVertex** `TO` **targetVertex** (both string|object): the two vertices between
```

Comment:
K_SHORTEST_PATH is not mentioned here - I think it should, even if it's only a keyword.

Reply:
Please add the Limit to the definition and detail description. Yes, there is a hint in the document - but users will just focus on the definition and run into trouble... 

```
FOR path 
  IN OUTBOUND|INBOUND|ANY K_SHORTEST_PATHS
  startVertex TO targetVertex
  GRAPH graphName
  [OPTIONS options]
  LIMIT <k>
```

Reply:
done.